plasmapy/dispersion/two_fluid_dispersion_solver.py

```diff
@@ -0,0 +1,243 @@
+    m_e : ~astropy.units.Quantity
+        Mass of negative ion which defaults to electron
+    m_i : ~astropy.units.Quantity
+        Mass of positive ion which defaults to proton
+    n : ~astropy.units.Quantity
+        Number density.
+    T_e : ~astropy.units.Quantity
+        The electron temperature
+    T_i : ~astropy.units.Quantity
+        The ion temperature
+    theta : ~astropy.units.Quantity
+        Angle of propagation which defaults to 45 degrees.
+    z : float or integer
+        Average ionization number which defaults to 1 for protons.
+
+    Returns
+    -------
+    omega : ~astropy.units.Quantity
+        A dictionary of Wave frequencies corresponding to three modes namely
+        a) Ion-acoustic mode, b) Alfven mode and c) Fast mode, in 1/s units.
+
+    Raises
+    ------
+    TypeError
+        The magnetic field and density arguments are not instances of
+        `~astropy.units.Quantity` and cannot be converted into those.
+
+    ~astropy.units.UnitConversionError
+        If the magnetic field or density is not in appropriate units.
+
+    ValueError
+        If the magnetic field, density or wavenumber is negative, or the ion
+        mass or charge state cannot be found.
+
+    Notes
+    -----
+    Computes the solution for wave dispersion relation based on equation 38 of
+    [1]
+
+    .. math::
+        \frac{\omega}{\omega_{ci}} = \sqrt(2\Lambda \sqrt(-\frac{p}{3}) cos
+        \left( \frac{1}{3} cos^{-1}\left( \frac{3q}{2p} \sqrt(-\frac{3}{p}
+        \right) - \frac{2\pi}{3}j \right) + \frac{A\Lambda}{3})
+
+    Where,
+        j = 0 ==> fast mode
+        j = 1 ==> Alfven mode
+        j = 2 ==> Acoustic mode
+
+    The above equation is derived from the general wave equation in the low
+    frequency regime, where both electrons and ions play significant role (in
+    the high frequency regime ions do not play any significant role and the
+    process is dominated by electron dynamics).
+
+    The complete dispersion equation is thus written as (from equation (1) of
+    [1]):
+
+    .. math::
+        \left( cos^2\theta - Q\frac{\omega^2}{k^2 {v_A}^2} \right) \left[
+        \left( cos^2\theta - \frac{\omega^2}{k^2 {c_s}^2 \right) -
+        Q\frac{\omega^2}{k^2 {v_A}^2 \left( 1 - \frac{\omega^2}{k^2 {c_s}^2
+        \right) \right] = \left(1 - \frac{\omega^2}{k^2 {c_s}^2 \right)
+        \frac{\omega^2}{{\omega_{ci}}^2}cos^2\theta
+
+    Here,
+    .. math::
+        Q = 1 + k^2 c^2/{\omega_{pe}}^2
+    \omega_{ci} is the proton gyrofrequency
+
+    References
+    ----------
+    .. [1] PM bellan, Improved basis set for low frequency plasma
+    waves, 2012, JGR, 117, A12219, doi:10.1029/2012JA017856
+    .. [2] TE Stringer, Low-frequency waves in an unbounded
+    plasma, 1963, JNE, Part C, doi:10.1088/0368-3281/5/2/304
+    .. [3] Rogers, B. N.; Denton, R. E.; Drake, J. F. & Shay, M. A.
+    Role of Dispersive Waves in Collisionless Magnetic Reconnection, prl, 2001,
+    87, 195004
```

Comment:
```suggestion
    .. [1] PM bellan, Improved basis set for low frequency plasma waves, 2012,
           JGR, 117, A12219, doi: `10.1029/2012JA017856
           <https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2012JA017856>`_.

    .. [2] TE Stringer, Low-frequency waves in an unbounded plasma, 1963, JNE,
           Part C, doi: `10.1088/0368-3281/5/2/304
           <https://doi.org/10.1088/0368-3281/5/2/304>`_

    .. [3] Rogers, B. N.; Denton, R. E.; Drake, J. F. & Shay, M. A. Role of
           Dispersive Waves in Collisionless Magnetic Reconnection, prl, 2001,
           87, 195004, doi: `10.1103/PhysRevLett.87.195004
           <https://doi.org/10.1103/PhysRevLett.87.195004>`_
```

On line continuation the text needs to be indented.  Added links for the DOI references.